libs/utils/utils.js

```diff
@@ -1023,29 +1025,72 @@ export async function loadMartech({
   }
 
   window.targetGlobalSettings = { bodyHidingEnabled: false };
-  loadIms().catch(() => {});
+  loadIms().catch(() => { });
 
   const { default: initMartech } = await import('../martech/martech.js');
   await initMartech({ persEnabled, persManifests, postLCP });
 
   return true;
 }
 
+/**
+ * Checks if the user is signed out based on the server timing and navigation performance.
+ * 
+ * @returns {boolean} True if the user is signed out, otherwise false.
+ */
+function isSignedOut() {
+  let w = window, perf = w.performance, serverTiming = {};
+
+  if (perf && perf.getEntriesByType) {
+    serverTiming = Object.fromEntries(
+      perf.getEntriesByType("navigation")?.[0]?.serverTiming?.map?.(
+        ({ name, description }) => ([name, description])
+      ) ?? []
+    );
+  }
+
+  const isSignedOutOnStagingOrProd = serverTiming && serverTiming.sis === '0';
+
+  // Return true if it's a dev environment or signed out on staging/prod
+  return !Object.keys(serverTiming || {}).length || isSignedOutOnStagingOrProd;
+}
+
+/**
+ * Enables personalization (V2) for the page.
+ * 
+ * @returns {boolean} True if personalization is enabled, otherwise false.
+ */
+export function enablePersonalizationV2() {
+  const enablePersV2 = document.head.querySelector(`meta[name="personalization-v2"]`);
+  return enablePersV2 && isSignedOut();
+}
+
 async function checkForPageMods() {
   const {
     mep: mepParam,
     mepHighlight,
     mepButton,
     martech,
   } = Object.fromEntries(PAGE_URL.searchParams);
+  let targetInteractionPromise = null;
   if (mepParam === 'off') return;
   const pzn = getMepEnablement('personalization');
   const promo = getMepEnablement('manifestnames', PROMO_PARAM);
   const target = martech === 'off' ? false : getMepEnablement('target');
   const xlg = martech === 'off' ? false : getMepEnablement('xlg');
+
   if (!(pzn || target || promo || mepParam
     || mepHighlight || mepButton || mepParam === '' || xlg)) return;
-  if (target || xlg) {
+
+  const enablePersV2 = enablePersonalizationV2();
+  if (martech !== 'off' && (target || xlg || pzn) && enablePersV2) {
+    const { locale } = getConfig();
+    const { loadAnalyticsAndInteractionData } = await import('../martech/helpers.js');
```

Comment:
Can we actually do this in a non blocking way as it was before?

### Before
![image](https://github.com/user-attachments/assets/cfc34191-1c36-419b-85e4-93cdc69f6495)

### After
![image](https://github.com/user-attachments/assets/9d459f01-617e-4654-87d6-b8458958d592)

Reply:
You can create a Promise that will then import the helpers and gets the interaction call and only then resolves. This way you don't need to wait for the module to load before moving on to loading the personalization module

Reply:
Addressed